GameServerLib/Logic/GameObjects/AttackableUnits/AI/ObjAIBase.cs

```diff
@@ -563,6 +568,59 @@ public override void OnCollision(GameObject collider)
             }
         }
 
+        public bool RecalculateAttackPosition()
+        {
+            if(Target != null && TargetUnit != null && !TargetUnit.IsDead && GetDistanceTo(Target) < CollisionRadius && GetDistanceTo(TargetUnit.X, TargetUnit.Y) <= Stats.Range.Total)//If we are already where we should be, do not move.
+            {
+                return false;
+            }
+            var objects = _game.ObjectManager.GetObjects();
+            List<CirclePoly> UsedPositions = new List<CirclePoly>();
+            var isCurrentlyOverlapping = false;
+
+            var thisCollisionCircle = new CirclePoly((Target != null ? Target.GetPosition() : GetPosition()), CollisionRadius + 10, 20);
+            
+            foreach (var gameObject in objects)
+            {
+                var u = gameObject.Value as AttackableUnit;
```

Comment:
Single letter variables are frowned on. Renaming to "unit" would be more acceptable.